L.Control.Geonames.js

```diff
@@ -203,8 +206,13 @@ L.Control.Geonames = L.Control.extend({
             var li;
             response.geonames.forEach(function(geoname){
                 li = L.DomUtil.create('li', '', this._resultsList);
-                li.innerHTML = this._getName(geoname);
+                var neededname = this._getName(geoname);
+                li.innerHTML = neededname;
                 L.DomEvent.addListener(li, 'click', function(){
+                    //The user picks a location and it changes the search text to be that location
+                    this._input.value = neededname;
+                    //The result list disappears
+                    L.DomUtil.removeClass(this._resultsList, 'hasResults');
```

Comment:
Let's invert this, and add or remove a new class `hidden` to show or hide this list.